### DISCUSSION TITLE
Large XGBoost/CatBoost models: how big is too big?

### DISCUSSION THREAD

**Discussion by probit2011** [shap/shap#3897] [Q&A]
I've been trying to use SHAP on a large XGBClassifier/CatBoostClassifier multiclass model with 17 features and 9 classes. At a learning rate of 0.05, I get an optimal model (tuned using Optuna) with around 130 estimators; the model itself can be saved as a ~100MB JSON file. If I push the learning rate down to 0.005, the number of estimators exceeds 1000 and the model's size goes up to ~500MB.

When I try to create a TreeExplainer with either of these models, Python (or the Jupyter kernel) within my VSCode crashes unceremoniously, with no explanations. I've searched far and wide to find the cause of the problem, but couldn't get any good answers. I even tried to re-build a specific Conda environment just for this task, but I got the same result. That's when I started to think about the size of the model and its possible impact on SHAP. I then tried to run the League of Legends example (which generates a model with a size of only 641KB) and it worked! So, model size seems to be the issue.

So, what would be the maximum model size (in memory terms) that TreeExplainer (or any of SHAP's explainers) can digest?
